### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by turbaszek**
This PR adds pre-commit hooks. Also, I've unified the licenses and added the missing ones.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      - id: insert-license
+        name: Add license for all go files
+        files: \.go$
+        exclude: ^\.github/.*$|^vendor/.*$|^zz_generated.*$
+        args:
+          - --comment-style
+          - "//"
+          - --license-filepath
+          - license-templates/LICENSE.txt
+          - --fuzzy-match-generates-todo
```
Comment (mik-laj) [inline .pre-commit-config.yaml]: Can you also add yamllint?  You can do it in a separate PR.
https://github.com/apache/airflow/blob/master/.pre-commit-config.yaml#L164-L170

**Comment by turbaszek** [comment]
@potiuk @mik-laj @ashb does Github Actions work for this repo? I see that it works on my fork but not here...

**Comment by mik-laj** [comment]
@turbaszek Github Action must be enabled through a ticket in INFRA.

**Comment by mik-laj** [comment]
GitHub actions works for this repository: https://github.com/apache/airflow-on-k8s-operator/actions

**Comment by turbaszek** [comment]
> GitHub actions works for this repository: https://github.com/apache/airflow-on-k8s-operator/actions

I know, in closed PRs you can find my test one. I still don’t know why it doesn’t work on this PR (work on fork).

**Comment by mik-laj** [comment]
@turbaszek Github doesn't work on forks.

**Comment by turbaszek** [comment]
> @turbaszek Github Action doesn't work on forks. This PR is from apache repo.

Well, it works. I still don't get how the GA works... 
This works:
```yml
on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```
this not:
```yml
on: [push,  pull_request]
```

**Comment by turbaszek** [comment]
@mik-laj @potiuk now we've got both yamllint and Github Actions :)

**Comment by kaxil** [comment]
> > @turbaszek Github Action doesn't work on forks. This PR is from apache repo.
> 
> Well, it works. I still don't get how the GA works...
> This works:
> 
> ```yaml
> on:
>   push:
>     branches:
>       - master
>   pull_request:
>     branches:
>       - master
> ```
> 
> this not:
> 
> ```yaml
> on: [push,  pull_request]
> ```

If the Pull request contains the branch from this repo itself, it would work. If you create a PR from your fork it won't work with GA

**Comment by turbaszek** [comment]
> If the Pull request contains the branch from this repo itself, it would work. If you create a PR from your fork it won't work with GA

Got it! My assumption was that `on: [push, pull_request]` works always, no matter what. Anyway, we have running static checks on CI! Should we move on? @kaxil @mik-laj
